siliconcompiler/design.py

```diff
@@ -0,0 +1,535 @@
+        return self._set(fileset, 'define', value, typelist=[str, list])
+
+    def get_define(self, fileset: str) -> List[str]:
+        """Returns defined macros for a fileset.
+
+        Args:
+           fileset (str): Fileset name.
+
+        Returns:
+           list[str]: List of macro definitions
+
+        """
+        return self._get(fileset, 'define')
+
+    ##############################################
+    def set_undefine(self, fileset: str, value: str) -> List[str]:
+        """Undefines a preprocessor macro for a fileset.
+
+        Args:
+           fileset (str): Fileset name.
+           value (str or List[str]): Macro definition to undefine.
+        """
+        return self._set(fileset, 'undefine', value, typelist=[str, list])
+
+    def get_undefine(self, fileset: str) -> List[str]:
+        """Returns undefined macros for a fileset.
+
+        Args:
+           fileset (str): Fileset name.
+
+       Returns:
+           list[str]: List of macro (un)definitions
+
+        """
+        return self._get(fileset, 'undefine')
+
+    ###############################################
+    def set_libdir(self, fileset: str, value: str) -> List[str]:
+        """Sets dynamic library directories for a fileset.
+
+        Args:
+           fileset (str): Fileset name.
+           value (str or List[str]): Library directories.
+        """
+        return self._set(fileset, 'libdir', value, typelist=[str, list])
+
+    def get_libdir(self, fileset: str) -> List[str]:
+        """Returns dynamic library directories for a fileset.
+
+        Args:
+           fileset (str): Fileset name.
+
+        Returns:
+           list[str]: List of library directories.
+
+        """
+        return self._get(fileset, 'libdir')
+
+    ###############################################
+    def set_lib(self, fileset: str, value: str) -> List[str]:
+        """Sets list of dynamic libraries for a fileset.
+
+        Args:
+           fileset (str): Fileset name.
+           value (str or List[str]): Library directory names.
+
+        """
+        return self._set(fileset, 'lib', value, typelist=[str, list])
+
+    def get_lib(self, fileset: str) -> List[str]:
+        """Returns list of dynamic libraries for a fileset.
+
+        Args:
+           fileset (str): Fileset name.
+
+        Returns:
+           list[str]: List of libraries.
+
+        """
+        return self._get(fileset, 'lib')
+
+    ###############################################
+    def set_param(self, fileset: str, name: str, value: str) -> str:
+        """Sets a named design parameter for a fileset.
+
+        Args:
+            fileset (str): Fileset name.
+            name (str): Parameter name.
+            value (str): Parameter value.
+        """
+
+        if not isinstance(fileset, str):
+            raise TypeError("fileset key must be a string")
+
+        if not isinstance(value, str) or value is None:
+            raise TypeError("param value must be a string")
```

Comment:
this wont hurt, but would be caught by the schemas key error checking.

Reply:
so in that case, it should be removed right?

Reply:
yes, this can be removed